clang/lib/CIR/CodeGen/CIRGenStmt.cpp

```diff
@@ -328,6 +328,14 @@ mlir::LogicalResult CIRGenFunction::buildLabelStmt(const clang::LabelStmt &S) {
   // IsEHa: not implemented.
   assert(!(getContext().getLangOpts().EHAsynch && S.isSideEntry()));
 
+  // TODO: After support case stmt crossing scopes, we should build LabelStmt
+  //       and clean LexicalScope::IsInsideCaseNoneStmt.
+  for (auto *lexScope = currLexScope; lexScope;
+       lexScope = lexScope->getParentScope()) {
+    assert(!lexScope->IsInsideCaseNoneStmt &&
```

Comment:
What happens if you remove this code? Also, why doesn't it work to just walk the scope up until you find a switch?